report_thesis/src/sections/background.tex

```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
+\end{equation}
+where $\hat{y}_m$ is the prediction of the $m$-th weak learner and $\alpha_m$ is the weight assigned to the $m$-th weak learner.
+While there are various choices for weak learners, decision trees are a common choice\cite{James2023AnIS}.
+
+\subsubsection{Decision Trees}
+A decision tree is a supervised learning model that partitions data into subsets based on feature values, creating a tree structure.
+The goal is to create a tree that predicts the target variable by dividing the data into increasingly homogeneous subsets.
+Each internal node in the tree represents a decision based on a specific feature, while each leaf node represents a prediction for the target variable.
+The tree can make predictions for new data points by following a path from the root to a leaf node\cite{James2023AnIS}.
+
+\subsubsection{Extra Trees Regressor (ETR)}
+We give an overview of \gls{rf} based on \citet{James2023AnIS}.
+Then, we introduce the \gls{etr} model based on \citet{geurtsERF}.
+
+\gls{rf} is an ensemble learning method that improves the accuracy and robustness of decision trees by building multiple trees and combining their predictions.
+Each tree is trained on a random subset of the data using bootstrap sampling, where samples are drawn with replacement, meaning the same sample can be selected multiple times.
+This introduces variability and reduces overfitting, as some data points may appear multiple times while others may be omitted.
+By aggregating predictions from multiple trees, the model achieves better generalization and robustness.
+
+Mathematically, the prediction of a \gls{rf} model can be represented as:
+
+$$
+f(x) = \frac{1}{M} \sum_{m=1}^{M} f_m(x),
+$$
+
+where $f_m(x)$ is the prediction of the $m$-th tree, and $M$ is the total number of trees.
+
+\gls{etr} extends the \gls{rf} model by introducing additional randomness in the tree-building process, specifically through random feature selection and random split points.
+While \gls{rf} uses bootstrap sampling and selects the best split from a random subset of features to create a set of diverse samples, \gls{etr} instead selects split points randomly within the chosen features, introducing additional randomness.
+This process results in even greater variability among the trees, aiming to reduce overfitting and improve the model's robustness.
+As a tradeoff, \gls{etr} is less interpretable than a single decision tree, as the added randomness can introduce more bias than \gls{rf}.
+However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
+
+\subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
+In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
+\gls{gbr} is an ensemble learning method that sequentially construct multiple decision trees, each one correcting the errors of the previous one, using gradient descent and boosting techniques.
```

Comment:
Boosting is literally what you describe: sequentially constructing trees (one at a time...), each one correcting the errors of the previous one. That's the boosting technique it uses - there aren't more.

The innovation with gradient boosting is to use gradient descent to minimize a loss function (which could be for regression, classification, etc., as long as the function is continuous and differentiable).
We start with some initial model. We compute the gradients of the loss function, w.r.t. the prediction of the current ensemble (just the initial model atm.). Then we fit a _new_ learner to the gradients. Then we add that learner to the ensemble, weighing it by some factor (i.e. a learning rate). And we repeat that. That's the 'gradient descent' process here.

Reply:
I resolve these because they are handled by another PR.